tests/unstake.test.ts

```diff
@@ -78,26 +92,19 @@ describe('Unstaking Integration Test', () => {
         });
 
         await mine(10);
-        const unstakeReceipt = await publicClient.waitForTransactionReceipt({ hash: unstakeHash });
+        const unstakeReceipt = await publicClient.getTransactionReceipt({ hash: unstakeHash });
         expect(unstakeReceipt.status).toEqual('success');
-        const userSharesFinal: bigint = await vaultContract.getShares(USER_ADDRESS);
 
-        expect(userSharesFinal).toEqual(userSharesInitial + AMOUNT_TO_STAKE - AMOUNT_TO_UNSTAKE);
-        const assets: bigint = await pool.connector.eth.readContract({
-            abi: VaultABI,
-            address: VAULT_ADDRESS,
-            functionName: 'convertToAssets',
-            args: [userSharesFinal],
-        });
-        expect(assets).toEqual(AMOUNT_TO_STAKE - AMOUNT_TO_UNSTAKE);
+        const { assets: assetsAfterUnstaking } = await pool.getStakeBalanceForUser(VAULT_ADDRESS);
+        expect(assetsAfterUnstaking).toEqual(assetsAfterStaking - AMOUNT_TO_UNSTAKE);
```

Comment:
looking good, but doesn't it take in account the gas?

Reply:
no, the gas is paid from the wallet balance.

Reply:
nevermind, got it, it's the balance on the pool not in the user's wallet